services/blockchain-app-registry/shared/utils/fsUtils.js

```diff
@@ -0,0 +1,95 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const path = require('path');
+const fs = require('fs');
+
+const { Logger } = require('lisk-service-framework');
+
+const logger = Logger();
+
+const isDirectory = directoryPath => fs.lstatSync(directoryPath).isDirectory();
```

Comment:
Please implement the asynchronous version of `fs.lstatSync` and use it wherever applicable.